beacon-chain/blockchain/head_sync_committee_info.go

```diff
@@ -110,6 +112,9 @@ func (s *Service) domainWithHeadState(ctx context.Context, slot types.Slot, doma
 func (s *Service) getSyncCommitteeHeadState(ctx context.Context, slot types.Slot) (state.BeaconState, error) {
 	var headState state.BeaconState
 	var err error
+	mLock := mputil.NewMultilock(fmt.Sprintf("%s-%d", "syncHeadState", slot))
+	mLock.Lock()
```

Comment:
fyi there's another lock in `syncCommitteeHeadStateCache` and I think you already know that